src/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -68,19 +68,23 @@ function BoundaryModelDummyParticles(initial_density, hydrodynamic_mass,
              create_cache_model(initial_density, density_calculator)...,
              create_cache_model(correction, initial_density, NDIMS, n_particles)...)
 
-    if reference_particle_spacing > 0.0
-        # since reference_particle_spacing has to be set for surface normals to be determined we can do this here
+    # If the `reference_density_spacing` is set calculate the `ideal_neighbor_count`
+    if reference_particle_spacing > 0
+        # `reference_particle_spacing` has to be set for surface normals to be determined
         cache = (;
                  cache...,  # Existing cache fields
+                 ideal_neighbor_count=Int(ideal_neighbor_count(Val(ndims(smoothing_kernel)),
+                                                               reference_particle_spacing,
+                                                               compact_support(smoothing_kernel,
+                                                                               smoothing_length))),
                  colorfield_bnd=zeros(ELTYPE, n_particles),
```

Comment:
```suggestion
                 colorfield=zeros(ELTYPE, n_particles),
```
This is in the cache of the boundary model, so it clearly is the colorfield of the boundary.

Reply:
that doesn't work